Claim event rewards

Guzbyte-tech opened the pull request:
1. Modified the createEvent function to suite reward system.
2. Created  `registerParticipant`  function to register participants to an event
3. Created `setEventMerkleRoot` to allow the creator of event to set merkleRoot for reward validation to suite the EventNFT contract
4. Added Claim Reward Function to allow reward system
5. Implemented the reward system for first participant.
6. Added NatSpec Comment for the  functions
7. Added test for all functions.

<img width="1320" alt="image" src="https://github.com/user-attachments/assets/06cd4d9e-f07b-427e-9a53-e5ef1b464032" />

Ugo-X commented:
Please resolve conflicts

Guzbyte-tech commented:
> Please resolve conflicts

Conflicts Resolved

Ugo-X reviewed (changes requested):
Generally, the token issue is not completely done, so this would have to be on hold, because you will need to have both implementations of the nft and the token to be done so that your PR will reflect the two implementations

Ugo-X reviewed (changes requested):
You Should move the implementation of the function to the rewardmanager contract.

And also he should bin the event merkle tree implementation as we're no longer using that.

Guzbyte-tech commented:
> You Should move the implementation of the function to the rewardmanager contract.
> 
> And also he should bin the event merkle tree implementation as we're no longer using that.




* Integrated WorldID for user verification in the EventNFT contract.
* Ensures secure and verifiable participation and reward claims using ZK proofs.
* Implemented a new EventManager contract to delegate reward distribution to EventRewardManager.
* Created the claimNFTReward function for participants to securely claim their NFT rewards.
* Integrated proof verification and reward handling logic.
* Added "First Participant" logic to reward the first user participating in an event with a special bonus NFT and Token Reward.
* Wrote extensive test cases for the EventManager contract.
* Validated core functionalities, edge cases, and failure scenarios.
* Enhanced EventRewardContract to support token-based rewards in addition to NFTs.
* Included detailed test coverage for token reward functionality.
